src/main/java/com/bio4j/angulillos/titan/TitanUntypedGraph.java

```diff
@@ -31,79 +32,90 @@ default TitanEdge addEdge(TitanVertex from, EdgeLabelMaker edgeType, TitanVertex
   @Override
   default TitanVertex addVertex(VertexLabelMaker type) {
 
-    return (TitanVertex) titanGraph().addVertexWithLabel(type.getName());
+    return (TitanVertex) titanGraph().addVertex(type.getName());
   }
 
   @Override
   default <V> V getPropertyV(TitanVertex vertex, String property) {
 
-    return vertex.<V>getProperty(property);
+    return vertex.<V>property(property).value();
   }
 
   @Override
   default <V> void setPropertyV(TitanVertex vertex, String property, V value) {
 
-    vertex.setProperty(property, value);
+    vertex.property(property, value);
   }
 
   @Override
   default <V> V getPropertyE(TitanEdge edge, String property) {
 
-    return edge.<V>getProperty(property);
+    return edge.<V>property(property).value();
   }
 
   @Override
   default <V> void setPropertyE(TitanEdge edge, String property, V value) {
 
-    edge.setProperty(property, value);
+    edge.property(property, value);
   }
 
   @Override
   default TitanVertex source(TitanEdge edge) {
 
-    return edge.getVertex(com.tinkerpop.blueprints.Direction.OUT);
+    return edge.outVertex();
   }
 
   @Override
   default TitanVertex target(TitanEdge edge) {
 
-    return edge.getVertex(com.tinkerpop.blueprints.Direction.IN);
+    return edge.inVertex();
   }
 
   @Override
   default Stream<TitanEdge> outE(TitanVertex vertex, EdgeLabelMaker edgeType) {
 
-    Iterable<Edge> itb = vertex.getEdges(com.tinkerpop.blueprints.Direction.OUT, edgeType.getName());
-
-    return stream( itb ).map( e -> (TitanEdge) e);
+    return stream(
+      vertex.query()
+        .labels(edgeType.getName())
+        .direction(Direction.OUT)
+        .edges()
+    );
```

Comment:
Note that here and in the similar methods (in/out-E/V), I've changed usage of the Tinkerpop API to the Titan `vertex.query()`, because it returns `TitanVertex`/`TitanEdge`s and seems to be more specific (I think that `.labels()` should use vertex-centric index when possible).